standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -1926,7 +1926,7 @@ public void create_dataconnector(final DataConnector connector)
       } catch (NoSuchObjectException e) {
         // expected
       }
-
+      firePreEvent(new PreCreateDataConnectorEvent(connector, this));
```

Comment:
We are calling firePreevent for PreCreateDataConnectorEvent here and at L#1887 (create_dataconnector_core api). Also, create_dataconnector_core() API is a private method and is only being called in this method. So I don't think we would need this firePreEvent here.